test/collectionUtils/CollectionUtils.test.ts

```diff
@@ -0,0 +1,65 @@
+import ContainsExtensionPredicate from "./testImpls/ContainsExtensionPredicate";
+import CollectionUtils from "../../src/collectionUtils/CollectionUtils";
+import ExtractValueTransformer, {
+  ValueHolder,
+} from "./testImpls/ExtractValueTransformer";
+
+describe("CollectionUtils", () => {
+  describe("#select()", () => {
+    it("should handle an empty input array", () => {
+      const predicate = new ContainsExtensionPredicate("!");
+      const result: Array<string> = CollectionUtils.select([], predicate);
+      expect(result).toBeArrayOfSize(0);
+    });
+
+    it("should return elements that match a Predicate", () => {
+      const predicate = new ContainsExtensionPredicate("!");
+      const input = ["one", "two!", "three!", "four"];
+      const result: Array<string> = CollectionUtils.select(input, predicate);
+      expect(result).toStrictEqual(["two!", "three!"]);
+    });
+  });
+
+  describe("#selectRejected()", () => {
+    it("should handle an empty input array", () => {
+      const predicate = new ContainsExtensionPredicate("!");
+      const result: Array<string> = CollectionUtils.selectRejected(
+        [],
+        predicate
+      );
+      expect(result).toBeArrayOfSize(0);
+    });
+
+    it("should return elements that do not match a Predicate", () => {
+      const predicate = new ContainsExtensionPredicate("!");
+      const input = ["one", "two!", "three!", "four"];
+      const result: Array<string> = CollectionUtils.selectRejected(
+        input,
+        predicate
+      );
+      expect(result).toStrictEqual(["one", "four"]);
```

Comment:
Codacy found an issue: [Unsafe call of an any typed value.](https://app.codacy.com/gh/MeasureAuthoringTool/model-info-parser/pullRequest?prid=5966907)